[git-webkit canonicalize] Support multiple empty lines

JonWBedard opened the pull request:
#### adcaeb7d203d7d889777eeb1750d104406f6df02
<pre>
[git-webkit canonicalize] Support multiple empty lines
<a href="https://bugs.webkit.org/show_bug.cgi?id=238744">https://bugs.webkit.org/show_bug.cgi?id=238744</a>
&lt;rdar://problem/91246947 &gt;

Reviewed by Stephanie Lewis and Dewei Zhu.

* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/canonicalize/message.py:
(main): When committing a change with &apos;Patch by&apos; in it, SVN adds an
extra empty line in the commit message.
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/test/canonicalize_unittest.py:
(TestCanonicalize.test_git_svn_existing_merge_queue): Added.

Canonical link: <a href="https://commits.webkit.org/249197@main">https://commits.webkit.org/249197@main</a>
git-svn-id: <a href="https://svn.webkit.org/repository/webkit/trunk@292304">https://svn.webkit.org/repository/webkit/trunk@292304</a> 268f45cc-cd09-0410-ab3c-d52691b4dbfc
</pre>

sundiamonde reviewed (approved):
rs=me.  
When doing this kind of formatting shenanigans I like to put comments with examples of what the input looks like.  Makes it easier for people unfamiliar with the code to follow the adjustments

dewei-zhu commented on Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/canonicalize/message.py:
```diff
@@ -69,16 +69,28 @@ def main(inputfile, identifier_template):
     if lines[identifier_index]:
         lines.insert(identifier_index, '')
 
+    # Turn multiple empty lines at the end of a commit into a single one
```
Should we have a test case for this?

JonWBedard replied:
We do: https://github.com/WebKit/WebKit/pull/241/files#diff-a67d5e0f2139831f939f41a909f52b7e25de86f374eda2faabadec5e67759d93R213

dewei-zhu reviewed (approved):
r=me with comment

JonWBedard committed:
[git-webkit canonicalize] Support multiple empty lines
https://bugs.webkit.org/show_bug.cgi?id=238744
<rdar://problem/91246947>

Patch by Jonathan Bedard <jbedard@apple.com> on 2022-04-04
Reviewed by Stephanie Lewis and Dewei Zhu.

* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/canonicalize/message.py:
(main): When committing a change with 'Patch by' in it, SVN adds an
extra empty line in the commit message.
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/test/canonicalize_unittest.py:
(TestCanonicalize.test_git_svn_existing_merge_queue): Added.
